recipes-hipos/hipos-time/files/set-time.sh

```diff
@@ -1,19 +1,62 @@
 #!/bin/bash
 
-MACHINE_TYPE="$(/usr/sbin/hip-machinfo -a)"
+log() {
+	logger -t set-time.sh $1
+}
+
 SUB_TYPE="$(/usr/sbin/hip-machinfo -s)"
-VARIANT=$(. /etc/hydraip-devid; echo "${device:-recorder}")
+
+# reset error signaling
+rm -f /etc/drerrors/RTC.json
 
 if [ "$SUB_TYPE" != "dvmon" -a "$SUB_TYPE" != "ipcam" ]; then
-        date -u -s "$(drbcc --dev=/dev/ttydrbcc --cmd=getrtc | cut -d\  -f2,3)" ||
-        {  # in case of problems we log the commands
-                set -x
-                for try in 1 2 3 4 5; do
-                date -u -s "$(drbcc --dev=/dev/ttydrbcc --cmd=getrtc | cut -d\  -f2,3)" && break
-                done 2>&1 | logger -t $0
-                set -x
-        }
-else
-        date -s `cat /lib/systemd/systemd-timesyncd` 2> /dev/null        
+	date=$(drbcc --dev=/dev/ttydrbcc --cmd=getrtc | cut -d\  -f2,3)
+	if [ -z "$date" ]; then
+		# in case of problems we log the commands
+		set -x
+		for try in 1 2 3 4 5; do
```

Comment:
My IDE's ShellCheck extension says "try appears unused. Verify use (or export if used externally).shellcheck(SC2034)" and more.

You can "shellcheck" your code using https://www.shellcheck.net/

Click on the SCxxxx links there for further explanation.

Reply:
Yes it seems to be unused, it is the retry counter.